crates/project/src/project.rs

```diff
@@ -3201,6 +3201,51 @@ impl Project {
         let disk_based_diagnostics_progress_token =
             adapter.disk_based_diagnostics_progress_token.clone();
 
+        // Log LSP messages under experimental/serverStatus and display pop-up for errors
```

Comment:
We usually prefer to avoid regular comments that state what the code is doing: in this case it seems pretty obvious from the block below, and with the rustdoc on `ServerStatus`, it will be self-explanatory IMO.
So let's add that rustdoc and remove this comment?